examples/GLLinePlotItem.py

```diff
@@ -29,19 +29,16 @@
 gz.translate(0, 0, -10)
 w.addItem(gz)
 
-def fn(x, y):
-    return np.cos((x**2 + y**2)**0.5)
-
 n = 51
 y = np.linspace(-10,10,n)
 x = np.linspace(-10,10,100)
 for i in range(n):
-    yi = np.array([y[i]]*100)
-    d = (x**2 + yi**2)**0.5
+    yi = y[i]
+    d = np.hypot(x, yi)
     z = 10 * np.cos(d) / (d+1)
-    pts = np.vstack([x,yi,z]).transpose()
+    pts = np.column_stack([x, np.full_like(x, yi), z])
```

Comment:
The logic is this, if x has shape (N,), it is treated as (N,1). Other shapes are left as is. i.e. (1,N) would not be treated as a column.
If you look at the example at https://github.com/pyqtgraph/pyqtgraph/issues/1732#issue-863957932, it does work in an intuitive manner because both X[i,:] and X[:,i] end up with a shape (N,)